pytorch_lightning/loops/dataloader/evaluation_loop.py

```diff
@@ -225,18 +231,15 @@ def _evaluation_epoch_end(self, outputs: EPOCH_OUTPUT) -> None:
         model._current_dataloader_idx = None
 
         if self.trainer.testing:
-            if is_overridden("test_epoch_end", model):
-                model._current_fx_name = "test_epoch_end"
-                model.test_epoch_end(outputs)
-
+            self.trainer._call_lightning_module_hook("test_epoch_end", outputs)
         else:
-            if is_overridden("validation_epoch_end", model):
-                model._current_fx_name = "validation_epoch_end"
-                model.validation_epoch_end(outputs)
+            self.trainer._call_lightning_module_hook("validation_epoch_end", outputs)
 
     def _on_evaluation_epoch_end(self) -> None:
         """Runs ``on_{validation/test}_epoch_end`` hook."""
         hook_name = "on_test_epoch_end" if self.trainer.testing else "on_validation_epoch_end"
-        self.trainer.call_hook(hook_name)
-        self.trainer.call_hook("on_epoch_end")
+        self.trainer._call_callback_hooks(hook_name)
+        self.trainer._call_lightning_module_hook(hook_name)
+        self.trainer._call_callback_hooks("on_epoch_end")
```

Comment:
nit: a newline would be nice here
```suggestion

        self.trainer._call_callback_hooks("on_epoch_end")
```